core/src/main/java/org/apache/iceberg/util/ArrayUtil.java

```diff
@@ -59,4 +68,239 @@ private ArrayUtil() {
+   * @return an {@code int} array, {@code null} if null array input
+   * @throws NullPointerException if array content is {@code null}
+   */
+  public static int[] toPrimitive(final Integer[] array) {
+    if (array == null) {
+      return null;
+    } else if (array.length == 0) {
+      return EMPTY_INT_ARRAY;
+    }
+    final int[] result = new int[array.length];
+    for (int i = 0; i < array.length; i++) {
+      result[i] = array[i].intValue();
+    }
+    return result;
+  }
+
+  /**
+   * This code is borrowed from `org.apache.commons:commons-lang3`.
+   * <p>
+   * Converts an array of object Longs to primitives.
+   * <p>
+   * This method returns {@code null} for a {@code null} input array.
+   *
+   * @param array  a {@code Long} array, may be {@code null}
+   * @return a {@code long} array, {@code null} if null array input
+   * @throws NullPointerException if array content is {@code null}
+   */
+  public static long[] toPrimitive(final Long[] array) {
+    if (array == null) {
+      return null;
+    } else if (array.length == 0) {
+      return EMPTY_LONG_ARRAY;
+    }
+    final long[] result = new long[array.length];
+    for (int i = 0; i < array.length; i++) {
+      result[i] = array[i].longValue();
+    }
+    return result;
+  }
+
+  /**
+   * This code is borrowed from `org.apache.commons:commons-lang3`.
+   * <p>
+   * Converts an array of object Floats to primitives.
+   * <p>
+   * This method returns {@code null} for a {@code null} input array.
+   *
+   * @param array  a {@code Float} array, may be {@code null}
+   * @return a {@code float} array, {@code null} if null array input
+   * @throws NullPointerException if array content is {@code null}
+   */
+  public static float[] toPrimitive(final Float[] array) {
+    if (array == null) {
+      return null;
+    } else if (array.length == 0) {
+      return EMPTY_FLOAT_ARRAY;
+    }
+    final float[] result = new float[array.length];
+    for (int i = 0; i < array.length; i++) {
+      result[i] = array[i].floatValue();
+    }
+    return result;
+  }
+
+  /**
+   * This code is borrowed from `org.apache.commons:commons-lang3`.
+   * <p>
+   * Converts an array of object Doubles to primitives.
+   * <p>
+   * This method returns {@code null} for a {@code null} input array.
+   *
+   * @param array  a {@code Double} array, may be {@code null}
+   * @return a {@code double} array, {@code null} if null array input
+   * @throws NullPointerException if array content is {@code null}
+   */
+  public static double[] toPrimitive(final Double[] array) {
+    if (array == null) {
+      return null;
+    } else if (array.length == 0) {
+      return EMPTY_DOUBLE_ARRAY;
+    }
+    final double[] result = new double[array.length];
+    for (int i = 0; i < array.length; i++) {
+      result[i] = array[i].doubleValue();
+    }
+    return result;
+  }
+
+  /**
+   * This code is borrowed from `org.apache.commons:commons-lang3`.
```

Comment:
Could you move this comment so it is the last paragraph? It is important to note, but not as important as the usage docs.

Reply:
Updated.